src/ccutil/object_cache.h

```diff
@@ -95,7 +95,7 @@ class ObjectCache {
 
   void DeleteUnusedObjects() {
     std::lock_guard<std::mutex> guard(mu_);
-    for (auto i = cache_.size() - 1; i >= 0; i--) {
+    for (int i = (int)cache_.size() - 1; i >= 0; i--) {
```

Comment:
The new fix is not the right one (C type cast). I think using a reverse iterator would be better.

Reply:
reverse iterator version. It seems to work fine, but I don't know how to test it since I don't know how this class is used. It may be better to have you fix it again.
```c++
void DeleteUnusedObjects() {
  std::lock_guard<std::mutex> guard(mu_);
  for (auto it = cache_.rbegin(); it != cache_.rend(); ++it) { 
    if (it->count <= 0) {
      delete it->object;
      cache_.erase(it.base());
    }
  }
}
```

Reply:
`apiexample_test` now tests that code (see commit d4bcbbd6defd97aeb5dbd1fbd58977dbd19563bd). I think your suggested code looks good.

Reply:
... looks good, but does not work. Code built with sanitizers crashes.